### PR TITLE
Edit existing gcse qualifications

### PR DESCRIPTION
**Pull request by alexdesi**
### Context

This PR allows to edit the existing GCSE qualifications (maths/English/science).

### Changes proposed in this pull request
Basically I am using the same principles we are using for the other forms.

Note: inside the form, I check if the `qualification_id` param exists, in that case, I make an update to the qualification model.

Note2: This might need some refactoring, but at the moment, since we are on MVP-focused mode, we prefer to bring some small tech debt and go faster.

### Guidance to review

Check the scenario that this PR is covering: `Candidate submits their maths GCSE details and then update them' 

### Link to Trello card
https://trello.com/c/eLRbW1Xo/301-editing-adding-maths-english-and-science-gcse-information

### CODE REVIEW

File: spec/system/candidate_interface/candidate_entering_gcse_spec.rb [hunk 1]
```diff
@@ -16,8 +16,19 @@
     when_i_fill_in_grade_and_year
     and_i_click_save_and_continue
 
-    then_i_see_the_review_for_maths_gcse
-    and_i_see_correct_grade_and_awarded_year
+    then_i_see_the_review_page_with_correct_details
+
+    # Sub-scenario: Edit Type of Qualification
```
Comment (tijmenb) [inline spec/system/candidate_interface/candidate_entering_gcse_spec.rb]:
```suggestion

```

File: app/controllers/candidate_interface/gcse/details_controller.rb [hunk 2]
```diff
@@ -4,30 +4,23 @@ class Gcse::DetailsController < CandidateInterfaceController
...
 
     def update
-      current_application_qualification = ApplicationQualification.last
+      details_form = GcseQualificationDetailsForm.build_from_qualification(current_qualification)
 
-      details_form = GcseQualificationDetailsForm.new(
-        grade: params[:candidate_interface_gcse_qualification_details_form][:grade],
-        award_year: params[:candidate_interface_gcse_qualification_details_form][:award_year],
-      )
+      details_form.grade = params[:candidate_interface_gcse_qualification_details_form][:grade]
```
[the remark this answers is not in the document]
Comment (alexdesi) [reply app/controllers/candidate_interface/gcse/details_controller.rb]: Done

File: app/controllers/candidate_interface/gcse/details_controller.rb [hunk 3]
```diff
@@ -4,30 +4,23 @@ class Gcse::DetailsController < CandidateInterfaceController
...
 
       if @application_qualification
         redirect_to candidate_interface_gcse_review_path
       else
         @application_qualification = details_form
 
-        render :edit_details
+        render :edit
       end
     end
```
[the remark this answers is not in the document]
Comment (alexdesi) [reply app/controllers/candidate_interface/gcse/details_controller.rb]: done

**Comment by tijmenb** [comment]
Can you squash the commits before merging?

**Pull request merged**